daemon/daemon_unix.go

```diff
@@ -820,7 +820,7 @@ func configureKernelSecuritySupport(config *config.Config, driverName string) er
 			return nil
 		}
 
-		if driverName == "overlay" || driverName == "overlay2" {
+		if driverName == "overlay" || driverName == "overlay2" || driverName == "overlayfs" {
```

Comment:
This is likely to cause confusions to the end users.

I think the containerd snapshotter name should be mapped to something like `containerd://overlayfs` or `containerd:overlayfs`

Reply:
There was some discussion about this in a maintainers call; see https://github.com/rumpl/moby/pull/40#discussion_r938221864

> We discussed this in the maintainers meeting;
> 
> * concensus was that the preference is to re-use the `--storage-driver` flag, as the term is generic enough, and was originally chosen for that reason (apparently there had been some internal discussion at that time to get rid of the "layers" concept, and early implementations of "snapshotter-like approaches" were already being designed)
> * for _now_, we would switch between "containerd snapshotters" and "storage drivers" based on the `feature` flag that was added (but snapshotter would of course at some point become the default)
> * if that feature is enabled, we must have some mapping for backward compatibility on (already familiar) names that have an equivalent in containerd, so;
>   
>   * `overlay` -> `io.containerd.snapshotter.v1.overlayfs`
>   * `overlay2` -> `io.containerd.snapshotter.v1.overlayfs` (same)
>   * `zfs` -> `io.containerd.snapshotter.v1.zfs`
>   * etc.
> * besides the "shorthand" (`overlay2`) names, also accept the "fully qualified" names (e.g. `io.containerd.snapshotter.v1.overlayfs` can be used in the daemon config)
> * if no explicit snapshotter is defined in the daemon config we can use containerd's default
> * but, we should consider having a better automatic selection (similar to how we currently select the most suitable snapshotter) instead of containerd's (currently hardcoded) default; this could be by querying (if possible) the list of available snapshotter-plugins that were loaded, and pick the first available one (based on a priority list).
> * perhaps this could use a contribution to containerd, to make containerd itself also make a similar selection (TBD)

Reply:
Overall it surprised me a bit that containerd didn't accept the full `io.containerd.snapshotter.v1.overlayfs` (which could've been an alternative). 

That said; the point of view there was that both "old" and "new" (containerd-snapshotter) being supported was more of a transitional period, after which there would be only "one" option.

Is your concern the configuration side of things, or what's shown in `docker info`? (perhaps for the `docker info` case we could consider having a different presentation)

/cc @tianon @cpuguy83 @tonistiigi @corhere any thoughts?

Reply:
^^ I should mention that we dropped the `io.containerd.snapshotter.v1.` handling in that PR, as it wouldn't (currently) match what's accepted by containerd, but perhaps that's a part that could be re-considered (and perhaps added in containerd itself)

Reply:
> (perhaps for the `docker info` case we could consider having a different presentation)

SGTM

Reply:
The documentation will also have to emphasize that `overlayfs` is actually newer than `overlay2`

Reply:
Yes; documentation will be added specifically about the containerd snapshotter integration; once all bits are integrated here in upstream, we will also update the "storage driver" section to describe them.